server/chat-commands/room-settings.ts

```diff
@@ -821,6 +822,8 @@ export const commands: ChatCommands = {
 			return this.errorReply(`An error occured while renaming the room.`);
 		}
 		this.modlog(`RENAMEROOM`, null, `from ${oldTitle}`);
+		room.log.roomlogFilename = "";
```

Comment:
I was confused by what this line was doing, and after further investigation, this doesn't seem to be doing anything useful. It looks like all it does is force `setupRoomlogStream` to re-create the roomlog stream even if it hasn't changed filenames, which isn't useful.

```suggestion
```

Reply:
I might be wrong, though. Am I missing something?

Reply:
it's necessary due to this line: https://github.com/smogon/pokemon-showdown/blob/master/server/roomlogs.ts#L136 
Honestly i would prefer to exclude that line because to me it just looks gross but as a result of the filename already being set to the current date after the rename it just causes it to fail despite not actually writing any logs.

Reply:
![](https://media.discordapp.net/attachments/430399256390795264/711823340754698280/unknown.png)
this should show the issue better than how i'm explaining it